test/k8s-integration/main.go

```diff
@@ -317,30 +344,34 @@ func handle() error {
 	return nil
 }
 
-func generateTestSkip(normalizedVersion string) string {
+func generateGCETestSkip(clusterVersion string) string {
 	skipString := "\\[Disruptive\\]|\\[Serial\\]"
-	switch normalizedVersion {
-	// Fall-through versioning since all test cases we want to skip in 1.15
-	// should also be skipped in 1.14
-	case "1.13":
-		fallthrough
-	case "1.14":
-		fallthrough
-	case "1.15":
-		fallthrough
-	case "1.16":
-		// "volumeMode should not mount / map unused volumes in a pod" tests a
-		// bug-fix introduced in 1.17
-		// (https://github.com/kubernetes/kubernetes/pull/81163)
+	v := apimachineryversion.MustParseSemantic(clusterVersion)
+
+	// "volumeMode should not mount / map unused volumes in a pod" tests a
+	// (https://github.com/kubernetes/kubernetes/pull/81163)
+	if v.LessThan(apimachineryversion.MustParseSemantic("1.16.0")) {
 		skipString = skipString + "|volumeMode\\sshould\\snot\\smount\\s/\\smap\\sunused\\svolumes\\sin\\sa\\spod"
-		// Skip Snapshot tests pre 1.17
-		skipString = skipString + "|snapshot"
-		fallthrough
-	case "1.17":
-	case "latest":
-	case "master":
-	default:
 	}
+
+	if v.LessThan(apimachineryversion.MustParseSemantic("1.17.0")) {
+		skipString = skipString + "|VolumeSnapshotDataSource"
+	}
+	return skipString
+}
+
+func generateGKETestSkip(clusterVersion string, use_gke_managed_driver bool) string {
+	skipString := "\\[Disruptive\\]|\\[Serial\\]"
+	curVer := mustParseVersion(clusterVersion)
+	if (*curVer).lessThan(mustParseVersion("1.16.0")) {
+		skipString = skipString + "|volumeMode\\sshould\\snot\\smount\\s/\\smap\\sunused\\svolumes\\sin\\sa\\spod"
+	}
+
+	if (use_gke_managed_driver && (*curVer).lessThan(mustParseVersion("1.18.3-gke.0"))) ||
```

Comment:
Will this need to be updated soon to include a 1.17 bound? Maybe include a TODO

Reply:
I don't think so, PD CSI Snapshots are never going to be available from the managed driver before this 1.18 version. I don't think we care about testing non-managed versions on GKE (?)

Reply:
I think we should consider getting this into 1.17 since the general snapshots feature is beta in 1.17.

Reply:
In order to get a managed driver snapsnot test in 1.17 we'll have to add a case to deploy only the sidecar manually?

I guess 1.17 will be around long enough it's worth having a test for that combination.

Reply:
GKE 1.17.5-gke.9 will have the pd csi snapshot support. I will put in the change here.